services/multitenant/legacy.go

```diff
@@ -0,0 +1,19 @@
+package multitenant
+
+import (
+	"time"
+)
+
+type Legacy struct {
+	*MultitenantStatsT
+}
+
+func WithLegacyPickupJobs(stats *MultitenantStatsT) *Legacy {
+	return &Legacy{stats}
+}
+
+func (*Legacy) GetRouterPickupJobs(_ string, _ int, _ time.Duration, jobQueryBatchSize int, _ float64) (map[string]int, map[string]float64) {
```

Comment:
These two return types look like good examples why using custom types instead of generic types (maps, strings, ints, etc.) might be useful sometimes for us uninitiated reviewers!

1. A reviewer who hasn't come across this piece of code before, cannot possibly know what a `map[string]int` and a `map[string]float64` stand for.
2. There is no documentation on this or the interface's method about what this method is supposed to return.
3. Even if there was some documentation on the interface, it would be hard from this method to jump to the interface and read the documentation without trying too hard (there might be some magical shortcut in VSCode which can help you find interface methods implemented by this struct that I am unaware of).
4. If instead of these 2 maps, 2 specific types were created with meaningful names, the names of the types alone might be enough to give some valuable context. If these types were documented as well, it would be like living in GitHub reviewer's paradise!

Please consider all the above as my excuse for not being able to understand whether what you return here is valid or not!

Reply:
@atzoum  Do you think named return types( example :` jobPerWorkspace map[string]int , pickedCustomerLatencyMap map[string]int`) would make sense or generic types ?

Reply:
@cisse21 any kind of custom typing can be useful, depending on the context.
At least for well established business identifiers of a domain, e.g in our case `workspaceId`, `sourceId`, `destinationId`, `destinationType`, `jobsDbPrefix` etc. it is worthwhile using custom types. It is common for such identifiers to be used as method parameters, or in maps and other parametrized types, so the gains become apparent immediately, e.g.
It is more straightforward to have a 
```
map[workspaceId]map[destinationType]int
```
rather than a 
```
map[string]map[string]int
```

At a higher level, it might also make sense to create a custom type which encapsulates another parametrized type, e.g. 
```
type pendingEventsCounters map[workspaceId]map[destinationType]int
```
The above type discloses even more information about its intention. Only by looking at it you can understand that it is supposed to hold counters grouped by workspace id & destination type. So you don't really care about what is the actual name of the variable holding this type (if this is a return type it might not be named), you can rely only on the type itself to infer its purpose.

@lvrach what is your opinion?